tests/audit-scanner.bats

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bats
+
+setup() {
+  load common.bash
+  wait_pods -n kube-system
+}
+
+@test "[Audit Scanner] Install testing policies and resources" {
+	run kubectl get cronjob -A
+	assert_output -p audit-scanner
+
+	# Launch unprivileged pod
+	kubectl run nginx-unprivileged --image=nginx:alpine
+	kubectl wait --for=condition=Ready pod nginx-unprivileged
+
+	# Launch privileged pod
+	kubectl run nginx-privileged --image=registry.k8s.io/pause --privileged
+	kubectl wait --for=condition=Ready pod nginx-privileged
+
+	# Create a namespace to trigger a fail evaluation in the audit scanner
+	kubectl apply -f $RESOURCES_DIR/testing-audit-scanner-namespace.yaml
+
+	# Deploy some policy
+	apply_cluster_admission_policy $RESOURCES_DIR/privileged-pod-policy.yaml
+	apply_cluster_admission_policy $RESOURCES_DIR/namespace-psa-label-enforcer-policy.yaml
+	apply_cluster_admission_policy $RESOURCES_DIR/safe-labels-namespace.yaml
+
+}
+
+@test "[Audit Scanner] Trigger audit scanner job" {
+    run kubectl get jobs -A
+    refute_output -p 'testing'
+
+    run kubectl create job  --from=cronjob/audit-scanner testing  --namespace $NAMESPACE
+    assert_output -p "testing created"
```

Comment:
I think we should wait for the successful completion of the job.
We should ensure that the policies are active and uniquely reachable beforehand (we do with `apply_cluster_admission_policy`), and we should ensure there's no other job from the audit-scanner cronjob being run in the background, particularly from `tests/audit-scanner_installation.bats`

Reply:
> I think we should wait for the successful completion of the job.

I'm not waiting for the job. But I'm waiting for the report. See next text in the file. Is it not enough?

> we should ensure there's no other job from the audit-scanner cronjob being run in the background, particularly from `tests/audit-scanner_installation.bats` , as it could influence the test results.

This is not possible. We [forbid](https://github.com/kubewarden/helm-charts/blob/25c0b7b193d4b453df23f550477c22eed161dd01/charts/kubewarden-controller/templates/cronjob.yaml#L13C3-L13C20) concurrent job. Furthermore, the `audit-scanner_installation.bats` will not run together with the `audit-scanner.bats`. As those tests are destructible, they are run separately as we do for the upgrade tests.

Reply:
Ok, I see that tests are not run in parallel, so `audit-scanner-installation.bats` shouldn't influence the outcome, correct?
I still think that we should add a wait for a successful completion of the job.

Reply:
I think it would be cleaner to wait for the report and assert the results, to be fair.

Reply:
> Ok, I see that tests are not run in parallel, so `audit-scanner-installation.bats` shouldn't influence the outcome, correct?

As far I know, by the default, `make` will run targets [sequentially](https://www.gnu.org/software/make/manual/html_node/Parallel.html). The users must define the `-j` to run them in parallel. But for us here, I think this is not necessary. Furthermore, the `audit-scanner-installation.bats` is never run together with the other tests. It can destroy the environment. Thus, the use must call this target manually. As we do in the CI script. In other words, the `audit-scanner-intallation.bats` should not influence the outcome of the `audit-scanner.bats` tests.

Reply:
> I still think that we should add a wait for a successful completion of the job.

Ok, I'll add a verification

Reply:
I've added the `kubectl wait` to wait for the `Complete` condition in the job. :)